src/taxpasta/infrastructure/application/ganon/ganon_profile_reader.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2023 Moritz E. Beber
+# Copyright (c) 2023 Maxime Borry
+# Copyright (c) 2023 James A. Fellows Yates
+# Copyright (c) 2023 Sofia Stamouli.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Provide a reader for ganon profiles."""
+
+
+import pandas as pd
+from pandera.typing import DataFrame
+
+from taxpasta.application.service import BufferOrFilepath, ProfileReader
+from taxpasta.infrastructure.helpers import raise_parser_warnings
+
+from .ganon_profile import GanonProfile
+
+
+class GanonProfileReader(ProfileReader):
+    """Define a reader for ganon profiles."""
+
+    @classmethod
+    @raise_parser_warnings
+    def read(cls, profile: BufferOrFilepath) -> DataFrame[GanonProfile]:
+        """
+        Read a ganon taxonomic profile from the given source.
+
+        Args:
+            profile: A source that contains a tab-separated taxonomic profile generated
+                by ganon.
+
+        Returns:
+            A data frame representation of the ganon profile.
+
+        """
+        result = pd.read_table(
+            filepath_or_buffer=profile,
+            sep="\t",
+            header=None,
+            index_col=False,
+            skipinitialspace=True,
+            names=[
+                GanonProfile.rank,
+                GanonProfile.target,
+                GanonProfile.lineage,
+                GanonProfile.name,
+                GanonProfile.nr_unique,
+                GanonProfile.nr_shared,
+                GanonProfile.nr_children,
+                GanonProfile.nr_cumulative,
+                GanonProfile.pc_cumulative,
+            ],
```

Comment:
So the ganon tool doesn't provide any column headers itself? How did you come up with the names for them, then?

Reply:
No ti does not, we took it from the output documentation: https://pirovc.github.io/ganon/outputfiles/#ganon-report

(replacing # with nr, and % with pc)

Reply:
Would you be very opposed to making them more expressive, like `nr` -> `number` and `pc` -> `percent`?

Reply:
No not at all! Go ahead